### PR TITLE
feat: Engagement Center Manage Rule - MEED-723 - Meeds-io/MIPs#13

### PR DESCRIPTION
**Pull request by AzmiTouil**
This PR will allow admin to manage rules in engagement center (add/update/delete)

### CODE REVIEW

**Review by SaraBoutej — COMMENTED** [review]
Review Added

File: portlets/src/main/webapp/js/RuleServices.js [hunk 1]
```diff
@@ -98,7 +98,7 @@ export function createRule(rule) {
     if (resp && resp.ok) {
       return resp.json();
     } else {
-      throw new Error('Error when updating rule');
+      throw new Error(resp.status);
```
Comment (SaraBoutej) [inline portlets/src/main/webapp/js/RuleServices.js:101]: throw new Error('Response code indicates a server error', resp);

File: portlets/src/main/webapp/vue-app/engagement-center/components/EngagementCenter.vue [hunk 2]
```diff
@@ -115,7 +122,13 @@ export default {
       this.tab = 1;
     } else if  (urlPath.indexOf(`${eXo.env.portal.context}/${eXo.env.portal.portalName}/contributions/achievements`) > -1) {
       this.tab = 2;
-    } 
+    }
+    this.$ruleServices.getEvents()
+      .then(events => {
+        this.events = events;
```
Comment (SaraBoutej) [inline portlets/src/main/webapp/vue-app/engagement-center/components/EngagementCenter.vue]: this.events = events || []

File: portlets/src/main/webapp/vue-app/engagement-center/components/challenges/Challenges.vue [hunk 3]
```diff
@@ -204,30 +206,32 @@ export default {
...
-          });
-        promises.push(retrieveChallengePromise);
-      }, 10);
+    if (urlPath.indexOf(`${eXo.env.portal.context}/${eXo.env.portal.portalName}/contributions/challenges`) > -1) {
+      this.challengeIdFromUrl = urlPath.match( /\d+/ ) && urlPath.match( /\d+/ ).join('');
+      if (this.providedId) {
+        setTimeout(() => {
+          const retrieveChallengePromise = this.$challengesServices.getChallengeById(this.providedId)
+            .then(challenge => {
+              if (challenge && challenge.id) {
```
Comment (SaraBoutej) [inline portlets/src/main/webapp/vue-app/engagement-center/components/challenges/Challenges.vue]: challenge?.id

File: portlets/src/main/webapp/vue-app/engagement-center/components/programs/ProgramDetail.vue [hunk 4]
```diff
@@ -111,6 +129,16 @@ export default {
       type: Object,
       default: null
     },
+    events: {
+      type: Array,
+      default: function () {
```
Comment (SaraBoutej) [inline portlets/src/main/webapp/vue-app/engagement-center/components/programs/ProgramDetail.vue]: default: () => [],

File: portlets/src/main/webapp/vue-app/engagement-center/components/programs/ProgramsList.vue [hunk 5]
```diff
@@ -59,7 +60,13 @@ export default {
     isAdministrator: {
       type: Boolean,
       default: false,
-    }
+    },
+    events: {
+      type: Array,
+      default: function() {
```
Comment (SaraBoutej) [inline portlets/src/main/webapp/vue-app/engagement-center/components/programs/ProgramsList.vue]: default: () => [],

**Review by SaraBoutej — CHANGES_REQUESTED** [review]

File: portlets/src/main/webapp/js/RuleServices.js (hunk 1 above)
Comment (AzmiTouil) [reply portlets/src/main/webapp/js/RuleServices.js:101]: I want to handle the error and display it by status

File: portlets/src/main/webapp/vue-app/engagement-center/components/challenges/Challenges.vue (hunk 3 above)
Comment (AzmiTouil) [reply portlets/src/main/webapp/vue-app/engagement-center/components/challenges/Challenges.vue]: done!

File: portlets/src/main/webapp/vue-app/engagement-center/components/programs/ProgramDetail.vue (hunk 4 above)
Comment (AzmiTouil) [reply portlets/src/main/webapp/vue-app/engagement-center/components/programs/ProgramDetail.vue]: done!

File: portlets/src/main/webapp/js/RuleServices.js (hunk 1 above)
Comment (SaraBoutej) [reply portlets/src/main/webapp/js/RuleServices.js:101]: ok

File: portlets/src/main/webapp/vue-app/engagement-center/components/programs/ProgramsList.vue (hunk 5 above)
Comment (AzmiTouil) [reply portlets/src/main/webapp/vue-app/engagement-center/components/programs/ProgramsList.vue]: done!

File: portlets/src/main/webapp/vue-app/engagement-center/components/EngagementCenter.vue (hunk 2 above)
Comment (AzmiTouil) [reply portlets/src/main/webapp/vue-app/engagement-center/components/EngagementCenter.vue]: done!

**Review by SaraBoutej — APPROVED** [review]
LGTM

**Pull request merged**